tests/rbd_mirror/test_rbd_image_meta_mirroring.py

```diff
@@ -0,0 +1,102 @@
+
+        mirror1.create_pool(poolname=poolname)
+        mirror2.create_pool(poolname=poolname)
+        mirror1.create_image(imagespec=imagespec, size=config.get("imagesize"))
+        mirror1.config_mirror(mirror2, poolname=poolname, mode="image")
+        mirror1.enable_mirror_image(poolname, imagename, "journal")
+        mirror2.wait_for_status(poolname=poolname, images_pattern=1)
+        mirror1.benchwrite(imagespec=imagespec, io=config.get("io-total"))
+        mirror1.wait_for_status(imagespec=imagespec, state_pattern="up+stopped")
+        mirror2.wait_for_status(imagespec=imagespec, state_pattern="up+replaying")
+
+        # Create image and enable snapshot mirroring"
+        mirror1.create_image(imagespec=imagespec_1, size=config.get("imagesize"))
+        mirror1.enable_mirror_image(poolname, imagename_1, "snapshot")
+        mirror2.wait_for_status(poolname=poolname, images_pattern=2)
+        mirror1.benchwrite(imagespec=imagespec_1, io=config.get("io-total"))
+        mirror1.wait_for_status(imagespec=imagespec_1, state_pattern="up+stopped")
+        mirror2.wait_for_status(imagespec=imagespec_1, state_pattern="up+replaying")
+
+        key = config.get("key")
+        value = config.get("value")
+        for i in imagespec, imagespec_1:
+            # Add meta value
+            rbd1.image_meta(action="set", image_spec=i, key=key, value=value)
+            if i == imagespec_1:
+                mirror1.create_mirror_snapshot(i)
+            time.sleep(30)
+            pdb.set_trace()
+            # Verify value at secondary
+            if value != rbd2.image_meta(action="get", image_spec=i, key=key)[:-1]:
+                log.error(f"Meta value did not get mirrored on {i}")
+                return 1
+            pdb.set_trace()
+
+            # change metadata
+            rbd1.image_meta(action="set", image_spec=i, key=key, value="changed")
+            if i == imagespec_1:
+                mirror1.create_mirror_snapshot(i)
+            time.sleep(30)
+            pdb.set_trace()
+
+            # Verify value at secondary
+            if "changed" != rbd2.image_meta(action="get", image_spec=i, key=key)[:-1]:
+                log.error(f"Change in meta value did not get mirrored on {i}")
+                return 1
+
+            # Remove from primary
+            rbd1.image_meta(action="remove", image_spec=i, key=key)
+            if i == imagespec_1:
+                mirror1.create_mirror_snapshot(i)
+            time.sleep(30)
+            pdb.set_trace()
+            # Verify removal at secondary
+            if type(rbd2.image_meta(action="get", image_spec=i, key=key)) == str:
+                log.error(f"Meta key removal did not get mirrored on {i}")
+
+        return 0
+    except ValueError as ve:
+        log.error(
+            f"{kw.get('ceph_cluster_dict').values} has less or more clusters Than Expected(2 clusters expected)"
+        )
+        log.exception(ve)
+    except Exception as e:
+        log.exception(e)
```

Comment:
Could we also log the exception in ERROR level?

Reply:
Sure, noted down.
I have made the changes.